llvm/lib/SYCLLowerIR/SYCLFrameworkOptimization.cpp

```diff
@@ -0,0 +1,24 @@
+//=== SYCLFrameworkOptimization.cpp - Utility pass for framework optimization //
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/SYCLLowerIR/SYCLFrameworkOptimization.h"
+
+#include "llvm/IR/Module.h"
+#include "llvm/Pass.h"
+
+using namespace llvm;
+using namespace sycl;
+
+PreservedAnalyses AddDebugFuncAttrs::run(Module &M, ModuleAnalysisManager &) {
+  for (Function &F : M) {
+    F.addFnAttr(Attribute::NoInline);
```

Comment:
Sorry I don't understand what this pass is neded for. I thought in the front end now if you pass -O0 but enable sycl framework O0 optimization, we remove optnone/noinline from SYCL framework functions. Isn't the IR already in the desired state at initial LLVM IR creation time?